Mathlib/Data/Matrix/Composition.lean

```diff
@@ -142,15 +141,20 @@ theorem compRingEquiv_apply (M : Matrix I I (Matrix J J R)) :
 theorem compRingEquiv_symm_apply (M : Matrix (I × J) (I × J) R) :
     (compRingEquiv I J R).symm M = (comp I I J J R).symm M := rfl
 
-end Semiring
+instance (R) [MulOne R] [AddCommMonoid R] [Fintype I] [DecidableEq I] [IsStablyFiniteRing R] :
+    IsStablyFiniteRing (Matrix I I R) :=
+  ⟨fun n ↦ let f := MonoidHom.mk ⟨compRingEquiv (Fin n) I R, comp_one⟩ (map_mul _)
+    MonoidHom.isDedekindFiniteMonoid_of_injective f (compRingEquiv ..).injective⟩
+
+end AddCommMonoid
 
 section LinearMap
 
-variable (K : Type*) [CommSemiring K] [AddCommMonoid R] [Module K R]
+variable (R₀ : Type*) [Semiring R₀] [AddCommMonoid R] [Module R₀ R]
 
 /-- `Matrix.comp` as `LinearEquiv` -/
 @[simps!]
-def compLinearEquiv : Matrix I J (Matrix K L R) ≃ₗ[K] Matrix (I × K) (J × L) R where
+def compLinearEquiv : Matrix I J (Matrix K L R) ≃ₗ[R₀] Matrix (I × K) (J × L) R where
```

Comment:
This reuse of `K` is really hilarious ... apparently no one ever tried to use this for anything else.